### ISSUE TITLE
null ref in EntityTypeExtensions.DisplayName when including a simple property

### ISSUE THREAD

**Issue by SimonCropp** [dotnet/efcore#12614]
TLDR: here is a failing test that repro's the problem https://github.com/SimonCropp/EfIncludeNullRefIssue/blob/master/DataModel/Tests.cs#L9

Ef version 2.1.1

If you do an include on a simple property (ie something that is not mapped to another table). a null ref is thrown in `Microsoft.EntityFrameworkCore.Metadata.Internal.EntityTypeExtensions.DisplayName`

![image](https://user-images.githubusercontent.com/122666/42487764-cc365fda-8445-11e8-8336-96fca185f18e.png)

it seems `IncludeCompiler.WalkNavigations` is using `longestMatchFound.Value.DisplayName()` when `Value` can be null

![image](https://user-images.githubusercontent.com/122666/42487851-3c07549a-8446-11e8-9164-154a426156c7.png)

### Exception

```
System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=Microsoft.EntityFrameworkCore
  StackTrace:
   at Microsoft.EntityFrameworkCore.Metadata.Internal.EntityTypeExtensions.DisplayName(IEntityType type) in C:\Code\EntityFrameworkCore\src\EFCore\Metadata\Internal\EntityTypeExtensions.cs:line 45
```

### Stacktrace
```

System.NullReferenceException : Object reference not set to an instance of an object.
at Microsoft.EntityFrameworkCore.Metadata.Internal.EntityTypeExtensions.DisplayName(IEntityType type) in C:\Code\EntityFrameworkCore\src\EFCore\Metadata\Internal\EntityTypeExtensions.cs:line 45
at Microsoft.EntityFrameworkCore.Query.Internal.IncludeCompiler.WalkNavigations(IEntityType entityType, IReadOnlyList`1 navigationPropertyPaths, IncludeLoadTree includeLoadTree) in C:\Code\EntityFrameworkCore\src\EFCore\Query\Internal\IncludeCompiler.cs:line 243
at Microsoft.EntityFrameworkCore.Query.Internal.IncludeCompiler.PopulateIncludeLoadTree(IncludeResultOperator includeResultOperator, IncludeLoadTree includeLoadTree) in C:\Code\EntityFrameworkCore\src\EFCore\Query\Internal\IncludeCompiler.cs:line 234
at Microsoft.EntityFrameworkCore.Query.Internal.IncludeCompiler.CreateIncludeLoadTrees(QueryModel queryModel) in C:\Code\EntityFrameworkCore\src\EFCore\Query\Internal\IncludeCompiler.cs:line 182
at Microsoft.EntityFrameworkCore.Query.Internal.IncludeCompiler.CompileIncludes(QueryModel queryModel, Boolean trackingQuery, Boolean asyncQuery) in C:\Code\EntityFrameworkCore\src\EFCore\Query\Internal\IncludeCompiler.cs:line 84
at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.OptimizeQueryModel(QueryModel queryModel, Boolean asyncQuery) in C:\Code\EntityFrameworkCore\src\EFCore\Query\EntityQueryModelVisitor.cs:line 326
at Microsoft.EntityFrameworkCore.Query.RelationalQueryModelVisitor.OptimizeQueryModel(QueryModel queryModel, Boolean asyncQuery)
at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.CreateQueryExecutor[TResult](QueryModel queryModel) in C:\Code\EntityFrameworkCore\src\EFCore\Query\EntityQueryModelVisitor.cs:line 170
at Microsoft.EntityFrameworkCore.Storage.Database.CompileQuery[TResult](QueryModel queryModel) in C:\Code\EntityFrameworkCore\src\EFCore\Storage\Database.cs:line 70
at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.CompileQueryCore[TResult](Expression query, IQueryModelGenerator queryModelGenerator, IDatabase database, IDiagnosticsLogger`1 logger, Type contextType) in C:\Code\EntityFrameworkCore\src\EFCore\Query\Internal\QueryCompiler.cs:line 126
at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.<>c__DisplayClass13_0`1.<Execute>b__0() in C:\Code\EntityFrameworkCore\src\EFCore\Query\Internal\QueryCompiler.cs:line 91
at Microsoft.EntityFrameworkCore.Query.Internal.CompiledQueryCache.GetOrAddQueryCore[TFunc](Object cacheKey, Func`1 compiler) in C:\Code\EntityFrameworkCore\src\EFCore\Query\Internal\CompiledQueryCache.cs:line 67
at Microsoft.EntityFrameworkCore.Query.Internal.CompiledQueryCache.GetOrAddQuery[TResult](Object cacheKey, Func`1 compiler) in C:\Code\EntityFrameworkCore\src\EFCore\Query\Internal\CompiledQueryCache.cs:line 44
at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.Execute[TResult](Expression query) in C:\Code\EntityFrameworkCore\src\EFCore\Query\Internal\QueryCompiler.cs:line 87
at Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryProvider.Execute[TResult](Expression expression) in C:\Code\EntityFrameworkCore\src\EFCore\Query\Internal\EntityQueryProvider.cs:line 62
at System.Linq.Queryable.FirstOrDefault[TSource](IQueryable`1 source)
at Tests.Foo() in C:\Code\EfIncludeNullRefIssue\DataModel\Tests.cs:line 22
```

**Comment by smitpatel** [comment]
Duplicate of #11971

**Comment by SimonCropp** [comment]
@smitpatel so just to be clear, in 2.2 the above scenario will no longer throw a null ref, but will instead throw a more descriptive exception?

**Comment by smitpatel** [comment]
@SimonCropp - Yes.

**Comment by smitpatel** [comment]
@SimonCropp - You can remove repro repository. We already have regression test checked in.
